xarray/tests/test_duckarray_testing.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+
+from .duckarray_testing import duckarray_module
+
+da = pytest.importorskip("dask.array")
+pint = pytest.importorskip("pint")
+sparse = pytest.importorskip("sparse")
+ureg = pint.UnitRegistry(force_ndarray_like=True)
+
+
+def create_pint(data, method):
+    if method in ("cumprod",):
+        units = "dimensionless"
+    else:
+        units = "m"
+    return ureg.Quantity(data, units)
+
+
+def create_pint_dask(data, method):
+    data = da.from_array(data, chunks=(2,))
+    return create_pint(data, method)
+
+
+def create_sparse(data, method):
+    return sparse.COO.from_numpy(data)
+
+
+TestPint = duckarray_module(
```

Comment:
I did consider that, but marks are indeed the reason why I decided against this. The mechanism to apply new marks would probably be similar (a class decorator accepting the `marks` dict), so that's not that much of an issue.

However, any mark applied to the base class will not be applied to the subclass, which means that we would have to override the methods just to reapply the marks (see the `test_pad` tests in `test_variable.py`).